pkg/utils/kubelessutil.go

```diff
@@ -53,9 +64,13 @@ func EnsureCronJob(client kubernetes.Interface, funcObj *kubelessApi.Function, s
 	eventTime := "\"event-time: $(date --rfc-3339=seconds --utc)\""
 	eventType := "\"event-type: application/json\""
 	eventNamespace := "\"event-namespace: cronjobtrigger.kubeless.io\""
-	commandTemplate := "curl -Lv -H %s -H %s -H %s -H %s %s"
+	headersTemplate := "-H %s -H %s -H %s -H %s"
+	headers := fmt.Sprintf(headersTemplate, eventId, eventTime, eventType, eventNamespace)
+
+	data := map[bool]string{true: fmt.Sprintf("-d %s ", payload), false: ""}[payload != "null"]
 
-	command := fmt.Sprintf(commandTemplate, eventId, eventTime, eventType, eventNamespace, functionEndpoint)
+	commandTemplate := "curl -Lv %s %s%s"
+	command := fmt.Sprintf(commandTemplate, headers, data, functionEndpoint)
```

Comment:
I think this is a bit more readable if you leave it as it is and you modify the value of command if the payload is not empty. Something like:

```
if payload != "" {
  command += fmt.Sprintf(" -d %s ", payload)
}
```
(you don't need to place all the args before the url)

BTW, doesn't the payload needs to be quoted to avoid wrong parsing of the JSON string? (I mean `-d '%s'`)

Reply:
Good idea, I'm changing the code to work that way :)